### PR TITLE
[4515] Correct customer id for context

### PR DESCRIPTION
**Pull request by riha112**
**Original issue:** 
* Fixes https://github.com/scandipwa/scandipwa/issues/4515

**Problem:** 
* Context for customer returns incorrect customer_id, due to context for session storing customer data after token revoking (after logout).

**In this PR:** 
* Passing authorized customer id if valid token passed in header.
* Revoking session.

### CODE REVIEW

**Comment by AleksandrsKondratjevs** [comment]
Maybe there is a sense to exclude session context get for setting user into main context at all?
I mean in this case we will always really only on token authorization context instead of both

File: src/Model/Context/AddUserInfoToContext.php [hunk 1]
```diff
@@ -57,21 +62,35 @@ class AddUserInfoToContext extends CoreAddUserInfoToContext
      */
     protected $dateTime;
 
+    /**
+     * @var Request
+     */
+    protected $request;
```
Comment (aleksandrsm) [inline src/Model/Context/AddUserInfoToContext.php:68]: class prop types

File: src/Model/Context/AddUserInfoToContext.php [hunk 2]
```diff
@@ -57,21 +62,35 @@ class AddUserInfoToContext extends CoreAddUserInfoToContext
...
+
+    /**
+     * @var TokenFactory
+     */
+    protected $tokenFactory;
```
Comment (aleksandrsm) [inline src/Model/Context/AddUserInfoToContext.php:73]: class prop types

File: src/Model/Context/AddUserInfoToContext.php [hunk 3]
```diff
@@ -129,6 +150,38 @@ public function execute(ContextParametersInterface $contextParameters): ContextP
         return $contextParameters;
     }
 
+    /**
+     * Returns authorized customer id; if valid token passed in header.
+     *
+     * @return int|null
+     */
+    protected function getCurrentUserId()
```
Comment (aleksandrsm) [inline src/Model/Context/AddUserInfoToContext.php:158]: maybe public? could see this being a target of a plugin

Comment (aleksandrsm) [reply src/Model/Context/AddUserInfoToContext.php:158]: also, if this is really int|null, can use ?int return type

File: src/Model/Resolver/RevokeCustomerToken.php [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+/**
+ * @inheritdoc
+ */
+class RevokeCustomerToken extends SourceRevokeCustomerToken
+{
+    /**
+     * @var Session
+     */
+    protected $session;
```
Comment (aleksandrsm) [inline src/Model/Resolver/RevokeCustomerToken.php:30]: class prop type

File: src/Model/Context/AddUserInfoToContext.php [hunk 5]
```diff
@@ -129,6 +150,38 @@ public function execute(ContextParametersInterface $contextParameters): ContextP
...
+    {
+        $authorizationHeaderValue = $this->request->getHeader(self::AUTH_HEADER);
+        if (!$authorizationHeaderValue) {
```
Comment (aleksandrsm) [inline src/Model/Context/AddUserInfoToContext.php:161]: newlines before ifs recommended

**Review by aleksandrsm — APPROVED** [review]
see minor code style changes

logic is fine

@AleksandrsKondratjevs 
I would not exclude setting of customer data in the session, because 3rd party tools rely on that. While they usually require FE adjustments to work with token auth instead of session, BE side can often work off of session without adjustments, there isn't a particular reason to break that.

**Comment by AleksandrsKondratjevs** [comment]
@aleksandrsm But no one says about removing data from session or session at all. I'm about removing it from build context.
Context right now as I know is constructed on top of 3 context depending on sortOrder and return value.
TokenAuthContext (10 sortorder)
CustomerSessionContext (default one magento session) (20 sortorder)
GuestContext (100 sortoder)

So when request is processed when it creates context it goes through those 3 context, one by one. If tokenauth return 0 for customer id and type then go to next one, if session is also 0 then go to gues.
Main issue out here is that TokenAuth return all correct, if user logouts we see that there is no token (basically this logic is from how auth token context processing request and getting out of it customer id and type), we can see that customer session is still exist and it start to get value from it.
So I though instead of starting to check auth token exist in request which is already done on auth token context, we could just remove from constructing context this one customer session.
It will still exist, it will still work and so on, it won't just pay any role on setting customer id and type into context for graphql request.

**Comment by aleksandrsm** [comment]
@AleksandrsKondratjevs 
It sounds like it might work, but I'm worried on the effect on 3rd party code. Not all graphql requests use token auth header in the first place. Even if that's a bad practice, removing session from the context building may break some things that were previously working.

I'd say it is better as is, with the session logout on token revoke. If only for backwards compatibility sake.

**Pull request merged**